lib/vfvalue.h

```diff
@@ -263,55 +266,24 @@ namespace ValueFlow
         /** The value bound  */
         Bound bound = Bound::Point;
 
-        /** int value (or sometimes bool value?) */
-        long long intvalue{};
-
-        /** token value - the token that has the value. this is used for pointer aliases, strings, etc. */
-        const Token* tokvalue{};
-
-        /** float value */
-        double floatValue{};
-
-        /** For calculated values - variable value that calculated value depends on */
-        long long varvalue{};
-
-        /** Condition that this value depends on */
-        const Token* condition{};
-
-        ErrorPath errorPath;
-
-        ErrorPath debugPath;
-
-        /** For calculated values - varId that calculated value depends on */
-        nonneg int varId{};
-
         /** value relies on safe checking */
-        bool safe{};
+        bool safe : 1;
 
         /** Conditional value */
-        bool conditional{};
+        bool conditional : 1;
 
         /** Value is is from an expanded macro */
-        bool macro{};
+        bool macro : 1;
 
         /** Is this value passed as default parameter to the function? */
-        bool defaultArg{};
+        bool defaultArg : 1;
 
-        int indirect{};
+        long long : 4;
```

Comment:
What the reason for this?

Reply:
That is explicit padding.

Reply:
explicit padding.. wait.. sounds like you are suboptimizing for a particular compiler now. Does this make the code faster on all compilers that is used to cppcheck? Will it make the code faster in future also.

Reply:
It gets rid of excessive padding and the explicit padding makes sure the warning gets triggered when this is modified in the future.

It will speed up things because less memory needs to be allocated (we did something similar with `Token` and `TokenImpl`).

But you are right that this adheres to the alignment to a specific platform (which is 8 bytes in this case). I do not think it should have that much impact on other platforms as only the explicit padding should affect them and that is kept quite low with 28 bits (which is way less than what we save).